autogen/sr.rs

```diff
@@ -38,97 +38,122 @@ impl OperandTokens {
         let name = get_param_name(operand);
         let iter = Ident::new(OPERAND_ITER, Span::call_site());
 
-        let (ty, lift_value) = match operand.kind.as_str() {
-            "IdRef" => match operand.name.trim_matches('\'') {
-                "Length" => (
-                    quote! { Token<Constant> },
-                    quote! { self.constants.lookup(*value).unwrap() },
-                ),
-                "Field Types" => (
-                    quote! { StructMember },
-                    quote! { types::StructMember::new(value.clone()) },
-                ),
-                "Parameter Types" => (
-                    quote! { Token<Type> },
-                    quote! { self.types.lookup(*value).unwrap() },
-                ),
-                name if name.ends_with(" Type") => (
-                    quote! { Token<Type> },
-                    quote! { self.types.lookup(*value).unwrap() },
-                ),
-                _ => (
-                    //TODO: proper `Token<>`
-                    quote! { spirv::Word },
-                    quote! { *value },
-                ),
+        let (ty, lift_value, first_name, second_name) = match operand.kind.as_str() {
+            "IdRef" => {
+                let (ty, value) = match operand.name.trim_matches('\'') {
+                    "Length" => (
+                        quote! { Token<Constant> },
+                        quote! { self.constants.lookup(*value).unwrap() },
+                    ),
+                    "Field Types" => (
+                        quote! { StructMember },
+                        quote! { types::StructMember::new(value.clone()) },
+                    ),
+                    "Parameter Types" => (
+                        quote! { Token<Type> },
+                        quote! { self.types.lookup(*value).unwrap() },
+                    ),
+                    name if name.ends_with(" Type") => (
+                        quote! { Token<Type> },
+                        quote! { self.types.lookup(*value).unwrap() },
+                    ),
+                    //TODO:
```

Comment:
Super nit: Typically we leave a space between `//` and the following text in comments in C++ world. Is this not true for Rust?

Reply:
Typically in Rust we are leaving a space after `//`. However, for special notes that start with TODO, FIXME, and so forth, I'm not putting the space in, which in part indicates that these comments are special. I'm fine with changing this, just explaining the general approach I've been doing in other projects.

Reply:
Sure. I don't care that much on this.